accumulo/src/main/java/com/v5analytics/simpleorm/AccumuloSimpleOrmSession.java

```diff
@@ -221,16 +232,20 @@ public void remove() {
     public <T> void alterVisibility(T obj, String currentVisibility, String newVisibility, SimpleOrmContext context) {
         try {
             ModelMetadata<T> modelMetadata = ModelMetadata.getModelMetadata(obj);
-            final Scanner scanner = createScanner(getTableName(modelMetadata), (AccumuloSimpleOrmContext) context);
-            scanner.setRange(new Range(modelMetadata.getId(obj)));
-            RowIterator rowIterator = new RowIterator(scanner);
-            BatchWriter writer = getBatchWriter(getTableName(modelMetadata));
-            ColumnVisibility newColumnVisibility = new ColumnVisibility(newVisibility);
-            while (rowIterator.hasNext()) {
-                Iterator<Map.Entry<Key, Value>> row = rowIterator.next();
-                alterVisibilityRow(writer, row, currentVisibility, newColumnVisibility);
+            Scanner scanner = createScanner(getTableName(modelMetadata), (AccumuloSimpleOrmContext) context);
+            try {
+                scanner.setRange(new Range(modelMetadata.getId(obj)));
+                RowIterator rowIterator = new RowIterator(scanner);
+                BatchWriter writer = getBatchWriter(getTableName(modelMetadata));
```

Comment:
does writer need to close?

Reply:
That writer came from [getBatchWriter](https://github.com/visallo/simple-orm/pull/13/files/cd17a0c2eace2bdde6e79ec482ae35025f738cd0#diff-511912cf7210184cde3a360a7eb62525R360) which means that it's been cached in the [batchWriters](https://github.com/visallo/simple-orm/pull/13/files/cd17a0c2eace2bdde6e79ec482ae35025f738cd0#diff-511912cf7210184cde3a360a7eb62525R34) map and shouldn't be closed.